pkg/registry/servicecatalog/binding/storage.go

```diff
@@ -139,6 +139,11 @@ func NewStorage(opts server.Options) (rest.Storage, rest.Storage, error) {
 		DestroyFunc: dFunc,
 	}
 
+	options := &generic.StoreOptions{RESTOptions: opts.EtcdOptions.RESTOptions, AttrFunc: GetAttrs}
+	if err := store.CompleteWithOptions(options); err != nil {
```

Comment:
`CompleteWithOptions` will set most of the fields in registry.Store for you, meaning you don't have to construct storage, key or name funcs, etc

```
	store := &genericregistry.Store{
		NewFunc:                  ...
		NewListFunc:              ...
		PredicateFunc:            ...
		DefaultQualifiedResource: ...

		CreateStrategy: ...
		UpdateStrategy: ...
		DeleteStrategy: ...
	}
	options := &generic.StoreOptions{RESTOptions: opts.EtcdOptions.RESTOptions, AttrFunc: GetAttrs}
	if err := store.CompleteWithOptions(options); err != nil {
		panic(err) // TODO: Propagate error up
	}
```

again, can be a follow up, but you really want to do this prior to having data you have to migrate, so you're sure you're consistent with resource paths in etcd, etc

Reply:
Yeah - I will do an immediate follow-up after this PR to simplify storage setup